metadata-io/src/main/java/com/linkedin/metadata/search/elasticsearch/update/ESBulkProcessor.java

```diff
@@ -109,6 +122,76 @@ public Optional<BulkByScrollResponse> deleteByQuery(QueryBuilder queryBuilder, b
         return Optional.empty();
     }
 
+    public Optional<Cancellable> reindex(QueryBuilder queryBuilder, String index, int batchSize, @Nullable TimeValue timeout) {
```

Comment:
Not sure this should be in the ESBulkProcessor. The other reindexing classes are with the index building/management classes in the indexbuilder package. Seems a bit out of place for this here.